app/controllers/backoffice/products_controller.rb

```diff
@@ -0,0 +1,72 @@
+module Backoffice
+  class ProductsController < ApplicationController
+    include AdminsOnly
```

Comment:
vc acha que dá pra explorar mais os belongs e os has_one no create e update

@category.products.build    ?

Reply:
no create sim, agora troquei graças ao refactor que vc sugeriu lá, vlw.
mas no update eu acredito que não é uma boa porque a pessoa pode trocar a categoria também no edit.

Então eu preciso:
- encontrar o produto pelo id
- encontrar a categoria pelo id, que pode ser diferente da categoria anterior do produto
- fazer o update com os parametros enviados, passando a nova (ou não) categoria como categoria.

não to vendo uma forma de fazer esse update em função da categoria

o que você acha @itsfernandaao ?

Reply:
vc não passa as duas variaveis para o edit? @category e @product ? ele não atualiza as duas e as retorna?

Reply:
como você faria?

Reply:
ele não modifica as variaveis né. é algo a se testar depois. mas manda brasa. poe logo o produto e passa os parametros. pq ja vem o category_id como parametro.

Reply:
eu incluiria category como parametro de produtos. agora nao sei se só por category. ou se poe category_id. pq de ele é um parametro de produto

Reply:
e ai nao precisa do merge. e nem precisa passar o objeto pro uptade.

Reply:
deixa eu ver uma referencia aqui, eu já fiz isso em algum lugar

Reply:
Beleza.
Se não me engano eu tentei a abordagem de passar só o id de category e ele reclama porque é uma instância de string e não de Category. Ele pede um objeto.

Reply:
def promotion_params
    params
      .require(:promotion)
      .permit(:name, :description, :code,
              :discount_rate, :coupon_quantity, :expiration_date,
              product_category_ids: [])
  end

Reply:
@emilyfaccin to be continued at the whats

Reply:
No caso que você listou é muitos pra muitos, é só adicionar o id na lista. Um pra muitos não rola. Ou eu to fazendo merda.
![image](https://user-images.githubusercontent.com/41349017/125776797-7fcb4184-6e00-4ea3-99ea-e31f338b4f1c.png)

Reply:
no final acho que é @product.update(product_params) , passando o category_id como product_params tb.